object_gomap.go

```diff
@@ -157,23 +157,68 @@ func (o *objectGoMapSimple) equal(other objectImpl) bool {
 	return false
 }
 
+// estimateMemUsage helps calculating mem usage for large objects.
+// It will sample the object and use those samples to estimate the
+// mem usage.
+func (o *objectGoMapSimple) estimateMemUsage(ctx *MemUsageContext) (estimate uint64, newEstimate uint64, err error) {
+	var samplesVisited, memUsage, newMemUsage uint64
+	counter := 0
+	totalProps := len(o.data)
+	sampleSize := totalProps / 10
+
+	// grabbing one sample every "sampleSize" to provide consistent
+	// memory usage across function executions
+	for key := range o.data {
+		counter++
+		if counter%sampleSize == 0 {
```

Comment:
[minor] if `totalProps` is < 10, then `sampleSize` == 0, when you hit `counter%sampleSize` you will get a `invalid operation: division by zero`.

Maybe only sample if `totalProps` is greater than 10, or maybe if `sampleSize` is 0 then set it to some arbitrary value?

Reply:
yup I think it's safer to have a guard here, will update with tests, thank you!